src/OVAL/oval_probe.c

```diff
@@ -329,14 +331,13 @@ int oval_probe_query_sysinfo(oval_probe_session_t *sess, struct oval_sysinfo **o
 	return(0);
 }
 
-static int oval_probe_query_criteria(oval_probe_session_t *sess, struct oval_criteria_node *cnode);
+static oval_result_t oval_probe_query_criteria(oval_probe_session_t *sess, struct oval_criteria_node *cnode, struct oval_result_criteria_node *rslt_cnode);
```

Comment:
@ybznek I think it can't be `const`, because subesequently we call functions `oval_criteria_node_get_test` and `oval_criteria_node_get_definition`. These functions are defined in public API but they don't have `const` in their prototypes. I have tried to add const modifier as you suggested, but it introduced some new compiler warnings. For example:

```
oval_probe.c:454:71: warning: passing argument 1 of ‘oval_criteria_node_get_definition’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  struct oval_definition *oval_def = oval_criteria_node_get_definition(cnode);
                                                                       ^
In file included from ./public/oval_probe.h:37:0,
                 from oval_probe.c:36:
./public/oval_definitions.h:2220:25: note: expected ‘struct oval_criteria_node *’ but argument is of type ‘const struct oval_criteria_node *’
 struct oval_definition *oval_criteria_node_get_definition(struct oval_criteria_node *);
```